docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile

```diff
@@ -15,10 +16,8 @@ WORKDIR /builds/parity/substrate/
 RUN set -eux; \
 	apt-get -y update; \
 	apt-get install -y --no-install-recommends \
-		software-properties-common curl git \
-		make cmake g++ clang rhash \
-		pkg-config libudev-dev time \
-		libssl-dev; \
+		git g++ clang cmake make \
+		pkg-config libssl-dev; \
```

Comment:
I completely agree with optimization but I consider *curl* like an essential package that can be useful in so many situations that I would leave it there (but I'm also fine having it removed).
**time** will certainly be needed for shell scripts that don't require bash.
@tomaka what about clang isn't it still needed in the builds of polkadot or substrate?

Reply:
`time` is built in, as well as `ca-certificates` and `gcc`.
OK, I'll include `curl`

Reply:
please verify on systems that don't have `/bin/sh` linked to bash e.g. Debian.

Reply:
actually:
```
The following NEW packages will be installed:
  clang clang-3.8 cmake cmake-data curl g++ g++-6 git git-man libarchive13
  libbsd0 libclang-common-3.8-dev libclang1-3.8 libcurl3 libcurl3-gnutls
  libdpkg-perl libedit2 liberror-perl libexpat1 libffi6 libgc1c2 libgdbm3
  libglib2.0-0 libgnutls30 libgssapi-krb5-2 libhogweed4 libicu57 libidn11
  libidn2-0 libjsoncpp1 libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0
  libldap-2.4-2 libldap-common libllvm3.8 liblzo2-2 libncurses5 libnettle6
  libnghttp2-14 libobjc-6-dev libobjc4 libp11-kit0 libperl5.24 libprocps6
  libpsl5 librtmp1 libsasl2-2 libsasl2-modules-db libssh2-1 libssl-dev
  libssl1.0.2 libstdc++-6-dev libtasn1-6 libunistring0 libuv1 libxml2 make
  perl perl-modules-5.24 pkg-config procps
```
`clang` is installed

Reply:
Okay sorry I was blind :see_no_evil: about the clang thing - sorry @tomaka nothing to see here :)

Reply:
should I move the changes to the branch with more appropriate name?

Reply:
I think the branch name won't be used anymore after merging. so for me it's fine.